### DISCUSSION TITLE
Why does the user talk to wings instead of the panel

### DISCUSSION THREAD

**Discussion by VasilisThePikachu** [pterodactyl/panel#4557] [General Community]
I have noticed this behavior before but i never thought about why it's like this. Is there a reason the user has to use websocket to communicate instead of having the panel make this communication?

**Comment by schrej** [comment]
The websocket is used to stream the console output to the user. It's just easier to implement it as a direct connection to the node, instead of forwarding it through the panel.
The two main disadvantages are that you need to setup ssl/https on the nodes (but you should do that anyway, even within a datacenter) and that the api of the node needs to be accessible from the internet. Since users will need to connect to the node anyway in order to connect to the game server, the latter isn't a major drawback either.

**Reply by VasilisThePikachu** [reply]
Thank you for the reply, my panel and wings are on the same machine so i was also thinking if some implementation like this would also perform faster and potentially safer(?) in my case. But this does explain why it's implemented like this
